src/L1/rollup/ScrollChain.sol

```diff
@@ -328,45 +428,33 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
     }
 
     /// @inheritdoc IScrollChain
-    /// @dev If the owner want to revert a sequence of batches by sending multiple transactions,
-    ///      make sure to revert recent batches first.
-    function revertBatch(bytes calldata _firstBatchHeader, bytes calldata _lastBatchHeader) external onlyOwner {
-        (
-            uint256 firstBatchPtr,
-            ,
-            uint256 _firstBatchIndex,
-            uint256 _totalL1MessagesPoppedOverallFirstBatch
-        ) = _loadBatchHeader(_firstBatchHeader);
-        (, , uint256 _lastBatchIndex, ) = _loadBatchHeader(_lastBatchHeader);
-        if (_firstBatchIndex > _lastBatchIndex) revert ErrorRevertZeroBatches();
-
-        // make sure no gap is left when reverting from the ending to the beginning.
-        if (committedBatches[_lastBatchIndex + 1] != bytes32(0)) revert ErrorRevertNotStartFromEnd();
+    function commitBatches(uint8 version, bytes32 lastBatchHash)
+        external
+        override
+        OnlySequencer
+        whenNotPaused
+        whenEnforcedBatchNotEnabled
+    {
+        _commitBatchesFromV7(version, lastBatchHash, false);
+    }
 
+    /// @inheritdoc IScrollChain
+    /// @dev This function cannot revert V6 and V7 batches at the same time, so we will assume all batches are V7.
+    /// If we need to revert V6 batches, we can downgrade the contract to the previous version and call this function.
+    /// @dev During commit batch we only store the last batch hash into storage. As a result, we cannot revert intermediate batches.
+    function revertBatch(bytes calldata batchHeader) external onlyOwner {
+        uint256 lastBatchIndex = miscData.lastCommittedBatchIndex;
+        (uint256 batchPtr, , uint256 startBatchIndex, ) = _loadBatchHeader(batchHeader, lastBatchIndex);
+        // only revert v7 batches
+        if (BatchHeaderV0Codec.getVersion(batchPtr) < 7) revert ErrorIncorrectBatchVersion();
         // check finalization
-        if (_firstBatchIndex <= lastFinalizedBatchIndex) revert ErrorRevertFinalizedBatch();
+        if (startBatchIndex < miscData.lastFinalizedBatchIndex) revert ErrorRevertFinalizedBatch();
 
         // actual revert
-        uint256 _initialEuclidBatchIndex = initialEuclidBatchIndex;
-        for (uint256 _batchIndex = _lastBatchIndex; _batchIndex >= _firstBatchIndex; --_batchIndex) {
-            bytes32 _batchHash = committedBatches[_batchIndex];
-            committedBatches[_batchIndex] = bytes32(0);
-
-            // also revert initial Euclid batch
-            if (_initialEuclidBatchIndex == _batchIndex) {
-                initialEuclidBatchIndex = 0;
-            }
-
-            emit RevertBatch(_batchIndex, _batchHash);
-        }
+        emit RevertBatch(startBatchIndex + 1, lastBatchIndex);
```

Comment:
If `startBatchIndex == lastBatchIndex` then this event will be a bit weird. But it's probably OK, we can treat any event where `startBatchIndex > finishBatchIndex` as a no-op.